File: Game.Api/Models/Skills/Skill.cs
using Game.Api.Models.Attributes;
using SkillEntity = Game.Core.Entities.Skill;

namespace Game.Api.Models.Skills
{
    public class Skill : IModelFromSource<Skill, SkillEntity>
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal BaseDamage { get; set; }
        public IEnumerable<AttributeMultiplier> DamageMultipliers { get; set; }
        public string Description { get; set; }
        public int CooldownMS { get; set; }
        public string IconPath { get; set; }

        public static Skill FromSource(SkillEntity skill)
        {
            return new Skill
            {
                Id = skill.Id,
                Name = skill.Name,
                BaseDamage = skill.BaseDamage,
                DamageMultipliers = skill.SkillDamageMultipliers.To().Model<AttributeMultiplier>(),
                Description = skill.Description,
                CooldownMS = skill.CooldownMs,
                IconPath = skill.IconPath,
            };
        }
    }
}

File: UI/new-svelte/src/lib/battle/skill.ts
import { IAttributeMultiplier, ISkill } from '$lib/api';
import { Battler } from './battler';

export class Skill implements ISkill {
	id: number;
	name: string;
	baseDamage: number;
	damageMultipliers: IAttributeMultiplier[];
	description: string;
	cooldownMs: number;
	iconPath: string;
	chargeTime = 0;
	renderChargeTime = 0;
	owner: Battler;

	constructor(data: ISkill, owner: Battler) {
		this.id = data.id;
		this.name = data.name;
		this.baseDamage = data.baseDamage;
		this.damageMultipliers = data.damageMultipliers;
		this.description = data.description;
		this.cooldownMs = data.cooldownMs;
		this.iconPath = data.iconPath;
		this.owner = owner;
	}

	public calculateDamage() {
		let dmg = this.baseDamage;
		this.damageMultipliers.forEach((dmgType) => {
			dmg += this.owner.attributes.getValue(dmgType.attributeId) * dmgType.multiplier;
		});
		return dmg;
	}
}
